lib/scope.js

```diff
@@ -1125,28 +1113,87 @@ module.exports = {
+      }
+    }
+
+    return enhanced_var_data;
+  },  // Ends scope.ensureSpecialRows()
+
+  getThereIsAnotherValue: async function ( scope, { row, from_table=true }) {
+    /* Given story row-like data as well as whether this is from a table row,
+    * return the best guess as to a safe value to set for this round. Give
+    * the developer a warning if
     * it's an inappropriate value. Avoid mutation.
     *
     * WARNING: Remember to decrement the actual value later.
+    * 
+    * Sometimes this will be from `.target_number` from story tables and
+    * we believe we've set it up so that people will make it a number.
+    * Sometimes this will be from `.there_is_another`. We think we've ensured
+    * that it will only come from Steps and should be `True` or `False`.
+    * This feels precarious, but I can't see another permuation.
+    * 
+    * This may in future be used to normalize all values that need `True`/`False`
+    * normalization.
     */
-    if ( parseInt( value ) < 1 || value === `False` || value === `false` ) {  // Loop will end
+    let value = row.value;
+    let int_val = parseInt( value );
+    
+    // Step stuff. Take into account a lot of possible values
+    if ( value.match( /(false|no|unchecked|uncheck|deselected|deselect)/i )) {
       return `False`;
 
-    } else if ( !from_table && ( value === `True` || value === `true` )) {
-      // This is not a table row, it's a step or something, so we should be safe from infinite loops
-      return `True`;
+    } else if ( value.match( /(true|yes|checked|check|selected|select)/i )) {
+      if ( from_table ) {
+        // Just in case, but this should never happen.
+        await scope.addToReport( scope, { key: `id`, value: `WARNING: The attribute \`.there_is_another\` is invalid in story table tests. Replace it with \`.target_number\` in your \`var\` and \`trigger\` columns. Set the \`value\` to the number of items in that list. This test will now set this row's \`value\` to \`False\`. See https://suffolklitlab.github.io/docassemble-AssemblyLine-documentation/docs/automated_integrated_testing#there_is_another. The row data is\n${ JSON.stringify( row.original )}` });
+        return `False`;
+      } else { return `True`; }
 
-    // Invalid value defaults to 'False' with a line added to the report
-    } else if ( isNaN( parseInt( value )) ) {  // Loop will end
-      // console.warn?
-      await scope.addToReport( scope, { key: `id`, value: `You must use a number or \`False\` in an \`.there_is_another\` row. The value was instead ${ row.value }. This test will default to \`False\` to avoid an infinite loop.` });
-      return `False`;
+    // Story table stuff
+    // Target: 3 (3 items. press 'yes' twice.)
+    // Times used: 0 -> `True`, 1 -> `True`, 2 -> `False`, 
+    } else if ( row.times_used >= int_val - 1 ) {  // Story table loop will end
```

Comment:
Should this check `row.source` as well?

Reply:
Thanks for spotting this! I think this only gets run right now at the beginning of the `processVar()` loop, so nothing has `source`, but I can see that it would be wise to future-proof it in case things change. I'll do that before we merge this.